test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -216,6 +217,7 @@ private void closeReader(EventStreamReader<Long> reader) {
             reader.close();
         } catch (Throwable e) {
             log.error("Error while closing reader", e);
+            Assert.fail("Error while closing reader. Test failure");
```

Comment:
I think you need to do
`testState.getReadException.set(e);` because this may be in another thread.

Reply:
done.